src/WebJobs.Script.WebHost/Management/FunctionsSyncManager.cs

```diff
@@ -0,0 +1,280 @@
+                    string currentHash = null;
+                    using (var md5 = MD5.Create())
+                    {
+                        byte[] hash = md5.ComputeHash(Encoding.UTF8.GetBytes(content));
+                        currentHash = hash
+                            .Aggregate(new StringBuilder(), (a, b) => a.Append(b.ToString("x2")))
+                            .ToString();
+                    }
+
+                    // get the last hash value if present
+                    string lastHash = null;
+                    if (await HashBlob.ExistsAsync())
+                    {
+                        lastHash = await HashBlob.DownloadTextAsync();
+                        _logger.LogDebug($"SyncTriggers hash (Last='{lastHash}', Current='{currentHash}')");
+                    }
+
+                    if (string.Compare(currentHash, lastHash) != 0)
+                    {
+                        // hash value has changed or was not yet stored
+                        // update the last hash value in storage
+                        await HashBlob.UploadTextAsync(currentHash);
+                        _logger.LogDebug($"SyncTriggers hash updated to '{currentHash}'");
+
+                        // only if we've updated the hash value should we return
+                        // true
+                        return true;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                // best effort
+                _logger.LogError(ex, "Error checking/updating SyncTriggers hash");
+            }
+
+            return false;
+        }
+
+        public async Task<JArray> GetSyncTriggersPayload()
+        {
+            var hostOptions = _applicationHostOptions.CurrentValue.ToHostOptions();
+            var functionsMetadata = WebFunctionsManager.GetFunctionsMetadata(hostOptions, _workerConfigs, _logger, includeProxies: true);
+
+            // Add trigger information used by the ScaleController
+            JObject result = new JObject();
+            var triggers = await GetFunctionTriggers(functionsMetadata, hostOptions);
+            return new JArray(triggers);
+        }
+
+        internal async Task<IEnumerable<JObject>> GetFunctionTriggers(IEnumerable<FunctionMetadata> functionsMetadata, ScriptJobHostOptions hostOptions)
+        {
+            var durableTaskConfig = await ReadDurableTaskConfig();
+            var triggers = (await functionsMetadata
+                .Where(f => !f.IsProxy)
+                .Select(f => f.ToFunctionTrigger(hostOptions))
+                .WhenAll())
+                .Where(t => t != null)
+                .Select(t =>
+                {
+                    // if we have a durableTask hub name and the function trigger is either orchestrationTrigger OR activityTrigger,
+                    // add a property "taskHubName" with durable task hub name.
+                    if (durableTaskConfig.Any()
```

Comment:
Ideally we'd figure out how to take this durable functions specific implementation out, but since it was already in here I agree it makes sense to leave it as is for the scope of this PR.